### DISCUSSION TITLE
Could this be another way to simulate photonic crystal waveguides?

### DISCUSSION THREAD

**Discussion by niubiworker** [NanoComp/meep#2919] [Q&A]
Can I try to simulate the transmission spectrum of this line defect waveguide and the photonic crystal band structure from the book Photonic Crystals: Molding the Flow of Light using Harminv in this example? This is because using the FDTD algorithm takes too long.
![屏幕截图 2024-10-20 220010](https://github.com/user-attachments/assets/abb7bb2d-45e4-4028-80c6-d3fbe82c7a83)
[]https://meep.readthedocs.io/en/latest/Python_Tutorials/Resonant_Modes_and_Transmission_in_a_Waveguide_Cavity/

**Comment by stevengj** [comment]
I'm not sure what you mean by "simulate the transmission spectrum".  Transmission from what to what?

The band structure and associated eigenfunctions (which were computed by MPB for this figure) are for *purely periodic* systems.  If you have an interface, e.g. with a different waveguide or a cavity or vacuum, then the periodic/Bloch eigenfunctions are generally not enough.

If you just want the band structure, use MPB.

**Reply by niubiworker** [reply]
My structure consists of a straight waveguide、coupled waveguide、line defect slow-light waveguide、coupled waveguide、straight waveguide, and I want to calculate the transmission from the straight waveguide to the straight waveguide.

I would also like to ask, if in the z-direction it's not simply air-dielectric-air, but for example, a thin film with a certain thickness is added on top of the line defect waveguide, in this case, the symmetry in the z-direction is broken, and it seems that the problem can no longer be easily solved using symmetry. How should I obtain the TE/TM band structure in this situation?

thank you very much!
